File: Segunda Practica Auxiliatura /Agregacion-Composicion/Eje1/Eje1.py
# INCISO A

class Habitacion:
    def __init__(self, nombre="", tamano=0.0):
        self._nombre = nombre
        self._tamano = tamano  

    def get_nombre(self):
        return self._nombre

    def get_tamano(self):
        return self._tamano

    def set_nombre(self, nombre):
        self._nombre = nombre

    def set_tamano(self, tamano):
        self._tamano = tamano

    def mostrar_info(self):
        print(f"Habitación: {self._nombre}, Tamaño: {self._tamano} m²")


class Casa:
    def __init__(self, direccion=""):
        self._direccion = direccion
        self._habitaciones = []  

    def get_direccion(self):
        return self._direccion

    def set_direccion(self, direccion):
        self._direccion = direccion

    def agregar_habitacion(self, habitacion):
        self._habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Casa en dirección: {self._direccion}")
        print("Habitaciones:")
        for hab in self._habitaciones:
            hab.mostrar_info()


# INCISO B

casa = Casa("Av. Siempre Viva 123")
casa.agregar_habitacion(Habitacion("Sala", 20.5))
casa.agregar_habitacion(Habitacion("Cocina", 12.3))
casa.agregar_habitacion(Habitacion("Dormitorio Principal", 25.0))
casa.agregar_habitacion(Habitacion("Baño", 6.7))

# INCISO C

casa.mostrar_casa()

File: Segunda Practica Auxiliatura /Agregacion-Composicion/Eje1/Eje1.java
// INCISO A

import java.util.ArrayList;
import java.util.List;

class Habitacion {
    private String nombre;
    private double tamano; 

    public Habitacion() {
        this.nombre = "";
        this.tamano = 0.0;
    }

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getTamano() {
        return tamano;
    }

    public void setTamano(double tamano) {
        this.tamano = tamano;
    }

    public void mostrarInfo() {
        System.out.println("Habitación: " + nombre + ", Tamaño: " + tamano + " m²");
    }
}

class Casa {
    private String direccion;
    private List<Habitacion> habitaciones;

    public Casa() {
        this.direccion = "";
        this.habitaciones = new ArrayList<>();
    }

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    
    public String getDireccion() {
        return direccion;
    }

    public void setDireccion(String direccion) {
        this.direccion = direccion;
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("Casa en dirección: " + direccion);
        System.out.println("Habitaciones:");
        for (Habitacion hab : habitaciones) {
            hab.mostrarInfo();
        }
    }
}

// INCISO B y C

public class Main {
    public static void main(String[] args) {
        Casa casa = new Casa("Av. Siempre Viva 123");
        casa.agregarHabitacion(new Habitacion("Sala", 20.5));
        casa.agregarHabitacion(new Habitacion("Cocina", 12.3));
        casa.agregarHabitacion(new Habitacion("Dormitorio Principal", 25.0));
        casa.agregarHabitacion(new Habitacion("Baño", 6.7));

        // INCISO C
        casa.mostrarCasa();
    }
}
